tools/generator/Java.Interop.Tools.Generator.CodeGeneration/CodeGenerator.cs

```diff
@@ -324,6 +328,25 @@ public void WriteClassProperties (ClassGen @class, string indent)
 			}
 		}
 
+		protected void WriteImplementedProperties (IEnumerable<Property> targetProperties, string indent, bool isFinal, GenBase gen)
+		{
+			foreach (var prop in targetProperties) {
+				bool get_virt = prop.Getter.IsVirtual;
+				bool set_virt = prop.Setter == null ? false : prop.Setter.IsVirtual;
+				prop.Getter.IsVirtual = !isFinal && get_virt;
+				if (prop.Setter != null)
```

Comment:
Question/test case: what should happen if the getter & setter *differ* in their "virtual-ness"?

In the case of Java default interface methods, this shouldn't be possible: https://stackoverflow.com/questions/23453287/why-is-final-not-allowed-in-java-8-interface-methods

However, if this method can be called for *class* methods, that certainly *is* possible...and has actually happened in Android, via [`Activity`](https://developer.android.com/reference/android/app/Activity)'s [`getTitle()`](https://developer.android.com/reference/android/app/Activity.html#getTitle()) and [`setTitle()`](https://developer.android.com/reference/android/app/Activity.html#setTitle(java.lang.CharSequence)) methods:

```xml
<class abstract="false" deprecated="not deprecated" extends="android.view.ContextThemeWrapper" extends-generic-aware="android.view.ContextThemeWrapper" jni-extends="Landroid/view/ContextThemeWrapper;" final="false" name="Activity" static="false" visibility="public" jni-signature="Landroid/app/Activity;">
  <method abstract="false" deprecated="not deprecated" final="true" name="getTitle" jni-signature="()Ljava/lang/CharSequence;" bridge="false" native="false" return="java.lang.CharSequence" jni-return="Ljava/lang/CharSequence;" static="false" synchronized="false" synthetic="false" visibility="public" />
  <method abstract="false" deprecated="not deprecated" final="false" name="setTitle" jni-signature="(Ljava/lang/CharSequence;)V" bridge="false" native="false" return="void" jni-return="V" static="false" synchronized="false" synthetic="false" visibility="public">
    <parameter name="title" type="java.lang.CharSequence" jni-type="Ljava/lang/CharSequence;" />
  </method>
</class>
```

(Above XML fragments from `xamarin-android/bin/BuildDebug/api/api-28.xml.in`.)

Which is to say, consider:

```java
public class Activity {
    public final CharSequence getTitle() {return null;}
    public void setTitle(CharSequence value) {}
}
```

*Currently* -- arguably a bug! -- we *remove* the "`virtual`-ness", and emit:

```csharp
partial class Activity {
  public unsafe Java.Lang.ICharSequence TitleFormatted {
    get { /*performs non-virtual dispatch */}
    set { /* performs virtual dispatch! */ }
  }
}
```

Which is admittedly "wonky", but... that's what we've done.

That doesn't mean we should *continue* this bizarro functionality for default interface methods.  (Which, again, we arguably *can't*, because you can't have `final` interface methods.)

Rephrased: maybe perhaps we should reconsider this code idiom?

That said, in "reconsidering" we *also* need to maintain ABI compatibility with existing `Mono.Android.dll`, but there are various ways to work around that...

Reply:
This function was renamed from `WriteClassProperties` to `WriteImplementedProperties` and is now used by by both classes and interfaces, so no logic changes were made.  Due to merge conflicts the old method was not removed.  Added a commit that removes the old method.

I think if this is something we want to tackle we should do so in a separate issue.  I'm not sure how it is solvable other than not converting the methods into a property, or generating the property as we do today (without `virtual`) and additionally adding a `virtual SetTitle (CharSequence)` method.

Reply:
Yes, this should be a separate issue/PR.

As for a fix, i think the `get` methods should become a read-only property,  and the `set` method should remain a separate method:

```csharp
class Activity {
    public ICharSequence TitleFormatted {
      get {...}
    }
    public virtual void SetTitle (ICharSequence value) {...}
}
```

Which, again, would result in API breakage in Mono.Android.dll, but that can be "solved"/"worked around" via Metadata.  The above is a saner default.